Multipage selection

ewchow opened the pull request:
Took a swing at adding support for multi-page PDFs (resolves #168). I kept the PDF management logic somewhat generalized, in case support for other file types are necessary in the future (TIFFs perhaps). Also tried to keep the existing logic as untouched as I could.

Main changes in this PR:
- Added UI page navigation elements
  - Added _Page X of Y_ display under _Image_ tree
    ![image](https://user-images.githubusercontent.com/633897/71393121-00d03f00-25d9-11ea-90f1-ad8567f0f382.png)
    - Clicking on _Page X of Y_ redirects selection to _Image_
    - Only applicable to multi-page PDFs
  - Added _Previous Page_, _Next Page_, and _Go to page: X_ controls in the _Image_ tree widget
    ![image](https://user-images.githubusercontent.com/633897/71393138-1a718680-25d9-11ea-8877-eea8c4b0f396.png)
- Added multi-page support to axes, datasets, and measurements
  - Output JSON structure includes additional _page_ key for all 3 sets
  - Only sets associated with current page are displayed in the UI
- Added multi-page support for undoing/redoing image cropping
- Added multi-page support for saving/loading JSON files
- Added PDF support for saving/loading project files 

I also took the liberty to add a `.gitignore` file. Happy to make any changes you'd like. Let me know what you think.

Happy Holidays!

ankitrohatgi commented on app/javascript/core/axes/bar.js:
```diff
@@ -154,6 +154,10 @@ wpd.BarAxes = (function() {
 
         this.name = "Bar";
+
+        if (wpd.appData.isMultipage()) {
```
I'd like to keep `core` pieces independent of the higher level `appData` as much as possible. Can we keep `page` out of both `axes` and `dataset`?

ewchow replied:
Sure, I'll work on moving them out

ankitrohatgi commented:
Hey, thanks again for the contributions and happy holidays to you too!

This looks good for most part - just the `axes.page`, `ds.page` and `ms.page` seem like maybe they need another thought. I have just superficially scanned though for now - I will take a closer look in the next few days.

ewchow commented:
Hey Ankit, hope you're well!

Finally got around to finishing up the refactor. I've moved the calls to `appData` out of `core` and removed `axes.page`, `ds.page`, and `ms.page` from the object definitions. The page managing logic is now more centralized in `pageManager`. The saved JSON will programmatically include the page info if it exists, otherwise it'll be as it were before.

ankitrohatgi commented:
Hi @ewchow, thanks for making the changes I requested. The new version looks good!

I see you have a few conflicts to fix - if you can resolve those, we can merge this :)

ewchow commented:
Great! The conflicts are resolved now. All set, thanks!
